openquake/hazardlib/tests/gsim/base_site_test.py

```diff
@@ -0,0 +1,110 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2012-2020 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+import numpy
+import unittest
+
+from openquake.hazardlib import const
+from openquake.baselib.hdf5 import read_csv
+from openquake.hazardlib.imt import PGA, SA
+from openquake.hazardlib.gsim.base import get_mean_std, _get_poes_site, _get_poes
+from openquake.baselib.general import gettemp, DictArray
+from openquake.hazardlib.contexts import DistancesContext
+from openquake.hazardlib.site_amplification import Amplifier
+from openquake.hazardlib.tests.gsim.mgmpe.dummy import Dummy
+from openquake.hazardlib.gsim.boore_atkinson_2008 import BooreAtkinson2008
+from openquake.hazardlib.gsim.boore_2014 import BooreEtAl2014
+
+from openquake.hazardlib.site import ampcode_dt
+from openquake.hazardlib.site_amplification import AmplFunction
+from openquake.hazardlib.tests.site_amplification_function_test import ampl_func
+
+
+class GetPoesSiteTestCase(unittest.TestCase):
+    # Add check that IMT and sigma are the same
+    # Add check on IMTs
+
+    def setUp(self):
+
+        fname = gettemp(ampl_func)
+        df = read_csv(fname, {'ampcode': ampcode_dt, None: numpy.float64},
+                      index='ampcode')
+        self.df = AmplFunction(df)
+
+        # Set GMMs
+        gmmA = BooreAtkinson2008()
+        gmmB = BooreEtAl2014()
+
+        # Set parameters
+        dsts = [10., 15., 20., 30., 40.]
+        dsts = [10.]
+        imts = [PGA(), SA(1.0)]
+        sites = Dummy.get_site_collection(len(dsts), vs30=760.0)
+        self.mag = 5.5
+        rup = Dummy.get_rupture(mag=self.mag)
+        dists = DistancesContext()
+        dists.rjb = numpy.array(dsts)
+        dists.rrup = numpy.array(dsts)
+        self.rrup = dists.rrup
+
+        # Compute GM on rock
+        # Shape: 2 x 4 (distances) x 2 (IMTs) x 1 (GMMs)
+        self.meastd = get_mean_std(sites, rup, dists, imts, [gmmA])
+        # Shape: 2 x 4 (distances) x 2 (IMTs) x 2 (GMMs)
+        self.meastd = get_mean_std(sites, rup, dists, imts, [gmmA, gmmB])
+
+    def test01(self):
+        import time
+
+        fname = gettemp(ampl_func)
+        df = read_csv(fname, {'ampcode': ampcode_dt, None: numpy.float64},
+                      index='ampcode')
+        df.reset_index(drop=False, inplace=True)
+        af = AmplFunction.from_compact(df)
+
```

Comment:
The name `from_compact` is not clear to me. Why not `from_dframe`?

Reply:
Done